### PR TITLE
feat: add new pointcloud format

### PR DESCRIPTION
**Pull request by Aysenayilmaz**
## Description
After updating the RobotecRGLLidar library, I've added the updated library and new script by cherry-picking from _**(@msz-rai)**_ the PR _(https://github.com/tier4/AWSIM/pull/334)_ opened in the [TierIV AWSIM](https://github.com/tier4/AWSIM) repo.  

I tried with Autoware main branch  and there is no problem with localization. 

### Related links

- https://github.com/autowarefoundation/autoware.universe/pull/6996
- https://github.com/autowarefoundation/AWSIM-Labs/issues/114
- https://github.com/RobotecAI/RobotecGPULidar/issues/313
- https://github.com/tier4/AWSIM/pull/334

## Tests performed

Not applicable.

## Effects on system behavior

Not applicable.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by xmfcx** [comment]
Could you fix the precommit errors?
https://github.com/autowarefoundation/AWSIM-Labs/actions/runs/10059193737/job/27803894740?pr=122#step:5:115

File: ProjectSettings/ProjectSettings.asset [hunk 1]
```diff
@@ -693,7 +695,7 @@ PlayerSettings:
     tvOS: 1
   incrementalIl2cppBuild: {}
   suppressCommonWarnings: 0
-  allowUnsafeCode: 0
+  allowUnsafeCode: 1
```
Comment (mozhoku) [inline ProjectSettings/ProjectSettings.asset]:
```suggestion
  allowUnsafeCode: 0
```

**Review by mozhoku — CHANGES_REQUESTED** [review]

**Review by mozhoku — APPROVED** [review]

**Comment by mozhoku** [comment]
It works with autoware-universe/main

LGTM :+1:

File: Assets/AWSIM/Scripts/Sensors/LiDAR/PointCloudFormatLibrary.cs [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+                }},
+                // PointXYZIRCEADT format used by Autoware
+                {PointCloudFormat.PointXYZIRCEADT, new[]
+                {
+                    RGLField.XYZ_VEC3_F32,
+                    RGLField.INTENSITY_U8,
+                    RGLField.RETURN_TYPE_U8,
+                    RGLField.RING_ID_U16,
+                    RGLField.ELEVATION_F32,
+                    RGLField.AZIMUTH_F32,
```
Comment (xmfcx) [inline Assets/AWSIM/Scripts/Sensors/LiDAR/PointCloudFormatLibrary.cs:72]: Order of these 2 lines should be changed.

File: Assets/AWSIM/Scripts/Sensors/LiDAR/PointCloudFormatLibrary.cs [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+                    RGLField.DISTANCE_F32,
+                    RGLField.RETURN_TYPE_U8,
+                    RGLField.PADDING_8,
+                    RGLField.PADDING_16,
+                    RGLField.PADDING_32,
+                    RGLField.TIME_STAMP_F64
+                }},
+                // PointXYZIRCEADT format used by Autoware
+                {PointCloudFormat.PointXYZIRCEADT, new[]
```
Comment (xmfcx) [inline Assets/AWSIM/Scripts/Sensors/LiDAR/PointCloudFormatLibrary.cs:65]: PointXYZIRC**EA**DT -> PointXYZIRC**AE**DT

**Comment by xmfcx** [comment]
Related:
- https://github.com/autowarefoundation/autoware.universe/issues/9020
- https://github.com/autowarefoundation/AWSIM-Labs/issues/173

File: Assets/AWSIM/Scripts/Sensors/LiDAR/PointCloudFormatLibrary.cs [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+using System.Collections.Generic;
+using RGLUnityPlugin;
+
+namespace AWSIM
+{
+    public enum PointCloudFormat : UInt32
+    {
+        Pcl24,
+        Pcl48,
+        PointXYZIRCEADT,
```
Comment (xmfcx) [inline Assets/AWSIM/Scripts/Sensors/LiDAR/PointCloudFormatLibrary.cs:25]: This too needs to be changed to `PointXYZIRCAEDT`

File: Assets/AWSIM/Scripts/Sensors/LiDAR/RglLidarPublisher.cs [hunk 5]
```diff
@@ -13,99 +13,163 @@
...
-            }
+                topic = "lidar/pointcloud",
+                publish = true,
+                fieldsPreset = PointCloudFormat.Pcl24,
+            },
+            new PointCloud2Publisher()
+            {
+                topic = "lidar/pointcloud_ex",
+                publish = true,
+                fieldsPreset = PointCloudFormat.PointXYZIRCEADT,
```
Comment (xmfcx) [inline Assets/AWSIM/Scripts/Sensors/LiDAR/RglLidarPublisher.cs:154]: This needs to be changed to `PointXYZIRCAEDT`

**Pull request merged**